torch/nn/attention/flex_attention.py

```diff
@@ -751,9 +771,15 @@ def score_mod(
         key (Tensor): Key tensor; shape :math:`(B, H, S, E)`.
         value (Tensor): Value tensor; shape :math:`(B, H, S, Ev)`.
         score_mod (Optional[Callable]): Function to modify attention scores. By default no score_mod is applied.
-        block_mask (BlockMask): BlockMask object that controls the blocksparsity pattern of the attention.
-        scale (Optional[float]): Scaling factor applied prior to softmax. If
+        block_mask (Optional[BlockMask]): BlockMask object that controls the blocksparsity pattern of the attention.
+
+    Kwargs:
+        scale (float): Scaling factor applied prior to softmax. If
         none, the default value is set to :math`\frac{1}{\sqrt{E}}`
+        rows_guaranteed_safe (bool): Is it guaranteed that at least one value in each row is not masked out?
```

Comment:
This is actually slightly more subtle since for a lot of masks this is true but it is not necessarily true at the block tiling level. So I think that we should update this to something like, for all rows is it guarnteed into a (Q_BLOCK_SIZE, KV_BLOCK_SIZE) block that all rows have guaranteed values.. Honestly I wonder how easy it is very people to reason about this..

Reply:
Should it be ```(BLOCK_M, BLOCK_N)``` instead of ```(Q_BLOCK_SIZE, KV_BLOCK_SIZE)```? Correct me if I'm wrong.